backend/src/v2/driver/driver.go

```diff
@@ -810,10 +814,10 @@ func resolveInputs(ctx context.Context, dag *metadata.DAG, iterationIndex *int,
 			value, hasValue := inputs.GetParameterValues()[name]
 
 			// Handle when parameter does not have input value
-			if !hasValue && inputsSpec.GetParameters()[name].IsOptional == false {
+			if !hasValue && !inputsSpec.GetParameters()[name].IsOptional {
 				// When parameter is not optional and there is no input value, report error
 				return fmt.Errorf("no value provided for non-optional parameter %q", name)
-			} else if !hasValue && inputsSpec.GetParameters()[name].IsOptional == true {
+			} else if !hasValue && inputsSpec.GetParameters()[name].IsOptional {
```

Comment:
Removed redundant comparisons

Reply:
nit: I'm not sure if the change is necessary here. These are just two different ways to assert true or false. One may even argue the previous way is more readable.